### DISCUSSION TITLE
rbenv can't install 3.2.2 or any other versions on mac0S 14 m2 Pro

### DISCUSSION THREAD

**Discussion by ajen010** [rbenv/ruby-build#2362] [General]
The logfile from the failed build has the section below showing the error:

```
readline.c:1903:37: error: use of undeclared identifier 'username_completion_function'; did you mean 'rl_username_completion_function'?

                                    rl_username_completion_function);

                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

                                    rl_username_completion_function

readline.c:79:42: note: expanded from macro 'rl_username_completion_function'

# define rl_username_completion_function username_completion_function

                                         ^

/usr/local/opt/readline/include/readline/readline.h:494:14: note: 'rl_username_completion_function' declared here

extern char *rl_username_completion_function (const char *, int);

             ^

installing default libraries

ld: warning: ignoring duplicate libraries: '-lruby.3.2'

compiling init.c

ld: warning: ignoring duplicate libraries: '-lruby.3.2'

1 error generated.

make[2]: *** [readline.o] Error 1

make[1]: *** [ext/readline/all] Error 2
```

**Comment by cassiopagnoncelli** [comment]
same issue

**Comment by mislav** [comment]
Hi, it looks like you've migrated from an Intel machine. Please see this section of the wiki for cause and solution: https://github.com/rbenv/ruby-build/wiki#ruby-on-apple-silicon

**Reply by cassiopagnoncelli** [reply]
Not my case. Compiler fails due to openssl

**Reply by mislav** [reply]
@cassiopagnoncelli In case of openssl-related failures, maybe this could provide some tips? https://github.com/rbenv/ruby-build/wiki#missing-openssl

It sounds like your issue is unrelated to the error message mentioning `rl_username_completion_function` that this discussion is about, though, and you are welcome to ask in a new discussion as long as you provide some output and logs. Thanks!

**Reply by cassiopagnoncelli** [reply]
I do get `rl_username_completion_function` issue trying another version
